environments/production/common/redis.tf

```diff
@@ -0,0 +1,97 @@
+locals {
+  redis = toset([
+    "frontend",
+    "backend-uk",
+    "backend-xi",
+    "admin",
+    "signon"
+  ])
+}
+
+resource "aws_elasticache_subnet_group" "this" {
+  name       = "redis-subnet-group"
+  subnet_ids = data.terraform_remote_state.base.outputs.private_subnet_ids
+}
+
+# tfsec:ignore:aws-cloudwatch-log-group-customer-key
+resource "aws_cloudwatch_log_group" "redis_slow_lg" {
+  for_each          = local.redis
+  name              = "redis-${each.key}-${var.environment}-slow-lg"
+  retention_in_days = 30
+}
+
+# tfsec:ignore:aws-cloudwatch-log-group-customer-key
+resource "aws_cloudwatch_log_group" "redis_engine_lg" {
+  for_each          = local.redis
+  name              = "redis-${each.key}-${var.environment}-engine-lg"
+  retention_in_days = 30
+}
+
+### Multi node redis with cluster mode disabled (one primary and two replica nodes spread across all 3 availability zones)
+module "redis" {
+  source   = "../../common/elasticache-redis/"
+  for_each = local.redis
+
+  replication_group_id        = "redis-${each.key}-${var.environment}"
+  description                 = "redis-${each.key}-${var.environment}"
+  parameter_group_name        = "default.redis7"
+  num_node_groups             = 1
+  replicas_per_node_group     = 2
+  node_type                   = "cache.t3.micro"
```

Comment:
I wonder if this will be ok for production?

Reply:
good catch `cache.t3.large` then?